### PR TITLE
graphviz: cleanup, use autoreconfHook, source from GitLab

### LINKED ISSUE #32708: graphviz download source is gone
## Issue description
```
trying http://www.graphviz.org/pub/graphviz/ARCHIVE/graphviz-2.40.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
error: cannot download graphviz-2.40.1.tar.gz from any mirror

```

### Steps to reproduce
Build graphviz on current master (1a98fc40e9bbfe6bdf17930b5fa926b3f50e5d27)

@bjornfor

### PR DESCRIPTION
**Pull request by lukateras**
###### Motivation for this change

Fixes #32708.

v2.32 doesn't have a proper tag (they only recently moved to Git), so instead sha1 is used as a `rev`, see: https://gitlab.com/graphviz/graphviz/commit/10c3c34c5198beacfba950764f34960c6884a34f

/cc @Fuuzetsu 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by yegortimoshenko** [comment]
Graphviz 2.32 doesn't have a proper tag (they only recently moved to Git), so instead sha1 is used as a `rev`, see: https://gitlab.com/graphviz/graphviz/commit/10c3c34c5198beacfba950764f34960c6884a34f

File: pkgs/tools/graphics/graphviz/base.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    substituteInPlace $out/bin/dotty --replace '`which lefty`' $out/bin/lefty
+    substituteInPlace $out/bin/vimdot \
+      --replace /usr/bin/vi '$(command -v vi)' \
+      --replace /usr/bin/vim '$(command -v vim)' \
+      --replace /usr/bin/vimdot $out/bin/vimdot \
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = https://graphviz.org;
+    description = "Praph visualization tools";
```
Comment (joachifm) [inline pkgs/tools/graphics/graphviz/base.nix]: Typo here, I think

**Comment by 7c6f434c** [comment]
Looks good modulo the typo.

**Comment by yegortimoshenko** [comment]
@7c6f434c Fixed. @joachifm, thanks for noticing!

**Pull request merged**